src/org/zaproxy/zap/extension/api/CoreAPI.java

```diff
@@ -964,6 +967,41 @@ public void process(HttpMessage msg) {
 			msg.setResponseBody(responseBody);
 			
 			return msg;
+		} else if (OTHER_SCRIPT_JS.equals(name)) {
+			String response = 
+					"function submitScript() {\n" +
+					"  var button=document.getElementById('button');\n" +
+					"  var component=button.getAttribute('zap-component')\n" +
+					"  var type=button.getAttribute('zap-type')\n" +
+					"  var name=button.getAttribute('zap-name')\n" +
+					"  var format\n" +
+					"  if (type == 'other') {\n" +
+					"    format = 'OTHER'\n" +
+					"  } else {\n" +
+					"    format = document.getElementById('zapapiformat').value\n" +
+					"  }\n" +
+					"  \n" +
+					"  var url = '/' + format + '/' + component + '/' + type + '/' + name + '/'\n" +
+					"  var form=document.getElementById('zapform');\n" +
+					"  form.action = url;\n" +
+					"  form.submit();\n" +
+					"}\n" +
+					"document.addEventListener('DOMContentLoaded', function () {\n" +
+					"  var button=document.getElementById('button');\n" +
+					"  if (button) {\n" +
+					"    document.getElementById('button').addEventListener('click',  function(e) {submitScript();}, false);\n" +
+					"  }\n" +
+					"});\n";
```

Comment:
Wouldn't it be more efficient to use a stringbuilder?

Actually should this be extracted to a constant or file instead of rebuilding it for every response where it's needed?